### PR TITLE
Signup: Remove radio's and button from `site-type` step

### PR DESCRIPTION
**Pull request by shaunandrews**
## Before
The current `site-type` step uses form elements like `radio` and `button` to let customers select a segment — It takes two clicks to pick a choice:

image

## After
We bet that making it easier, and quicker, to pick an option will lead to higher conversions. This PR updates the UI to be less complex, using clickable rows with a `chevron` icon to select segments:

image

## How to test

1. Visit `/start/onboarding/site-type`

### CODE REVIEW

**Comment by fditrapani** [comment]
@shaunandrews I made a little update that takes care of the functionality but it broke the layout/design a little in an effort to make the cards more semantic.

**Comment by shaunandrews** [comment]
Perf, thanks @fditrapani — all seems to work great now. Updated the CSS to work with the new markup.

**Comment by fditrapani** [comment]
Looks good to me! Should be good to merge once we get some dev eyes on it.

File: client/signup/steps/site-type/index.jsx [hunk 1]
```diff
@@ -35,8 +35,9 @@ class SiteType extends Component {
 			hasInitializedSitesBackUrl,
 		} = this.props;
 
-		const headerText = translate( 'Start with a site type' );
-		const subHeaderText = '';
+		const headerText = translate( 'What are we building today?' );
```
Comment (fditrapani) [inline client/signup/steps/site-type/index.jsx]: We should try keep changes like these in separate PRs incase changes for the radio buttons get caught up in reviews.

**Comment by fditrapani** [comment]
Looks like this change happens on Jetpack too (thanks @michaeldcain!). Screen Shot 2019-03-12 at 10 48 46 PM

@jeffgolenski @joanrho we'd like to make an update to the site segment component to make it quicker for folks to pick a choice. Next up we'll be adding some icons too p5XAZ9-27C-P2. Do you have any thoughts or concerns?

**Comment by dzver** [comment]
The code looks good and works fine, should be ready to merge after somebody from Jetpack takes a look.

**Comment by michaeldcain** [comment]
@tyxla @johnHackworth for Jetpack code review, since it modifies a shared component.

**Comment by jeffgolenski** [comment]
@fditrapani @shaunandrews I love this because it reduces a click / tap. Based on the screen recordings I watched for the jetpack flow when we implemented our steps, no one was second guessing their choices, so I think this is a much more efficient method for customers to move forward. Genius. 

cc'ing the jetpack OB team for good measure, @joanrho @keoshi @tyxla @AnnaMag @johnHackworth

File: client/signup/steps/site-type/index.jsx [hunk 2]
```diff
@@ -35,8 +35,9 @@ class SiteType extends Component {
...
+		const subHeaderText =
```
Comment (southp) [inline client/signup/steps/site-type/index.jsx]: We should wrap this in `translate()`

File: client/signup/steps/site-type/form.jsx [hunk 3]
```diff
@@ -40,52 +36,34 @@ class SiteTypeForm extends Component {
...
-		event.preventDefault();
-
+	handleSubmit = type => {
 		this.props.recordTracksEvent( 'calypso_signup_actions_submit_site_type', {
-			value: siteType,
+			value: type,
 		} );
 
-		this.props.submitForm( siteType );
+		this.setState( { siteType: type } );
```
Comment (southp) [inline client/signup/steps/site-type/form.jsx]: This `siteType` state was used by the radio buttons to track selection. Now that they are gone, it can be removed entirely, from all the `setState()` call and the constructor.

File: client/signup/steps/site-type/form.jsx [hunk 4]
```diff
@@ -40,52 +36,34 @@ class SiteTypeForm extends Component {
...
+		this.props.submitForm( type );
 	};
 
 	renderRadioOptions() {
 		return allSiteTypes.map( siteTypeProperties => (
-			<FormLabel
-				className={ classNames( 'site-type__option', {
-					'is-selected': siteTypeProperties.slug === this.state.siteType,
-				} ) }
+			<button
```
Comment (tyxla) [inline client/signup/steps/site-type/form.jsx]: I don't think we need to rediscover the wheel for this: we already have a linkable card component - see https://wpcalypso.wordpress.com/devdocs/design/card - the "I am a linkable card" and "I am a clickable button that looks like a link" examples.

File: client/signup/steps/site-type/form.jsx (hunk 3 above)
Comment (AnnaMag) [reply client/signup/steps/site-type/form.jsx]: We still need it to track selection for Jetpack sites.

**Comment by AnnaMag** [comment]
code review has been taken care of, so I tested the functionality for JP sites. LGTM.

File: client/signup/steps/site-type/index.jsx (hunk 1 above)
Comment (michaeldcain) [reply client/signup/steps/site-type/index.jsx]: Moved to #31467

File: client/signup/steps/site-type/index.jsx (hunk 2 above)
Comment (michaeldcain) [reply client/signup/steps/site-type/index.jsx]: Moved to #31467

**Comment by shaunandrews** [comment]
I added a `:focus` state (with a counterpart for JP) along with styling the icons for each segment.

image
image

image

**Comment by fditrapani** [comment]
Looking nice @shaunandrews.

**Comment by keoshi** [comment]
Looking real nice — thanks, Shaun! I'll review it live tomorrow morning.

File: client/signup/steps/site-type/form.jsx [hunk 5]
```diff
@@ -40,54 +36,34 @@ class SiteTypeForm extends Component {
...
+			value: type,
 		} );
 
-		this.props.submitForm( siteType );
+		this.setState( { siteType: type } );
+
+		this.props.submitForm( type );
 	};
 
 	renderRadioOptions() {
```
Comment (AnnaMag) [inline client/signup/steps/site-type/form.jsx]: Function name does not seem to be relevant anymore.

Comment (fditrapani) [reply client/signup/steps/site-type/form.jsx]: Good catch!

File: client/signup/steps/site-type/form.jsx [hunk 6]
```diff
@@ -40,54 +36,34 @@ class SiteTypeForm extends Component {
...
+	renderSegmentOptions() {
 		return allSiteTypes.map( siteTypeProperties => (
-			<FormLabel
-				className={ classNames( 'site-type__option', {
-					'is-selected': siteTypeProperties.slug === this.state.siteType,
-				} ) }
+			<Card
+				className="site-type__option"
 				key={ siteTypeProperties.id }
+				displayAsLink={ true }
```
Comment (tyxla) [inline client/signup/steps/site-type/form.jsx]: `={ true }` is unnecessary, just passing `displayAsLink` will make it `true` anyway.

File: client/lib/signup/site-type.js [hunk 7]
```diff
@@ -55,6 +58,7 @@ export const allSiteTypes = [
 	{
 		id: 4,
 		slug: 'education',
+		icon: 'premium-bundle',
```
Comment (tyxla) [inline client/lib/signup/site-type.js]: `premium-bundle` doesn't seem to be a valid icon, as it doesn't correspond to a valid plan slug. Did you mean `value_bundle`?

**Comment by ranh** [comment]
I think the pencil / rollerball / fountain pen metaphor made sense when we thought of our plans as good / better / best, but our plans don’t map to that scheme anymore.

Another issue with the icons is that we are using the same icons, and in some cases the same labels, to mean different things. We already know from testing that customers confuse site types with upgrade plans. Tying the two concepts together further by using the same icons is likely to increase the potential for confusion.

Perhaps we can use just the colors and leave the icons themselves out?

**Comment by shaunandrews** [comment]
I ditched the icons for now — if we wanna bring them back we'll need to find better icons.

image

**Comment by jeffgolenski** [comment]
@shaunandrews You're going to have to roll-back this change. It looks _too good_ 

Awesome work. Just tested with Jetpack's flow. It's fantastic. cc @keoshi @joanrho
